### PR TITLE
Avoid deleting files only on exit in some places

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/funTest/kotlin/ExamplesFunTest.kt [hunk 1]
```diff
@@ -159,6 +158,7 @@ class ExamplesFunTest : StringSpec() {
             )
 
             report shouldHaveSize 1
+            outputDir.deleteRecursively() shouldBe true
```
Comment (fviernau) [inline cli/src/funTest/kotlin/ExamplesFunTest.kt:161]: The added assertion is not related to asserting what the test is about.
I'd prefer to not use assertions for that, but rather a `require` if really needed.
What do you think?

Have you considered `safelyDeleteRecursively(force=true)` instead?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: cli/src/funTest/kotlin/ExamplesFunTest.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/funTest/kotlin/ExamplesFunTest.kt:161]: > The added assertion is not related to asserting what the test is about.

That depends a bit on how strict you want to be. To me, properly closing the file handle in `generateReport()` is something that makes sense to be tested in this context. And not closing the file handle properly would prevent deletion of the directory on Windows at least. So from that perspective, I'd argue it's fine to have this assertion here.

> I'd prefer to not use assertions for that, but rather a `require` if really needed.
> What do you think?

Besides my above argumentation, I'd prefer to not mix assertions with requirements in tests for clarity.

> Have you considered `safelyDeleteRecursively(force=true)` instead?

Good hint, done. Although it actually shouldn't be required here. But I still did that change for consistency with other places where we delete temporary directories in tests.

Comment (fviernau) [reply cli/src/funTest/kotlin/ExamplesFunTest.kt]: > That depends a bit on how strict you want to be. 

I find it easier to grasp what the test assertions are if it suffices to look at all `shoulds` without the need for filtering that out.
I think it's best to ignore that topic in scope of this PR.

**Review by fviernau — APPROVED** [review]

File: model/src/test/kotlin/config/NexusIqConfigurationTest.kt [hunk 2]
```diff
@@ -33,14 +33,14 @@ import org.ossreviewtoolkit.model.writeValue
...
         "support a serialization round-trip via an ObjectMapper" {
-            val referenceOrtConfig = OrtConfiguration.load(file = File("src/main/resources/reference.conf"))
-            val rereadOrtConfig = createTempFile(suffix = ".yml").toFile().apply {
-                writeValue(referenceOrtConfig)
-                deleteOnExit()
-            }.readValue<OrtConfiguration>()
+            val ortConfig = OrtConfiguration.load(file = File("src/main/resources/reference.conf"))
+            val rereadOrtConfig = createTempFile(suffix = ".yml").toFile().run {
+                writeValue(ortConfig)
+                readValue<OrtConfiguration>().also { delete() }
```
Comment (oheger-bosch) [inline model/src/test/kotlin/config/NexusIqConfigurationTest.kt:39]: I am not sure whether this kind of change is safe. If an exception is thrown by `readValue()` or the test fails for other reasons, the temporary file will not be deleted, will it?

It seems that Kotest has a feature to support [temporary files](https://kotest.io/docs/framework/temporary-files/). Would this be an alternative?

Comment (sschuberth) [reply model/src/test/kotlin/config/NexusIqConfigurationTest.kt:39]: > It seems that Kotest has a feature to support [temporary files](https://kotest.io/docs/framework/temporary-files/). Would this be an alternative?

That's indeed an alternative I was also looking into, but I didn't want to take it into use until there is a Kotest release which includes [my changes to their directory naming](https://github.com/kotest/kotest/pull/2140). But then again, maybe we should start using the feature already now, and only benefit from the improved naming on the next Kotest upgrade.

Comment (sschuberth) [reply model/src/test/kotlin/config/NexusIqConfigurationTest.kt:39]: Uh, I just discovered another issue with Kotest's `tempdir()` implementation that needs to be fixed first...

**Comment by sschuberth** [comment]
I'll rework this / split into other PRs.

**Pull request closed without merging**